bitcoin/src/amount.rs

```diff
@@ -1530,6 +1533,16 @@ mod tests {
 
     use super::*;
 
+    
+    #[test]
+    fn from_str_zero() {
+        let denoms = vec!["BTC", "mBTC", "uBTC", "nBTC", "pBTC", "bits", "sats", "msats"];
+        for denom in denoms {
+            let s = format!("0 {}", denom);
+            assert!(Amount::from_str(&s).is_ok(), "Failed to create amount from {}", s);
+        }
+    }
+
```

Comment:
Oh I didn't realise we were supposed to support strange things like "-0" and "000", this test fails
```rust
    #[test]
    fn from_str_zero() {
        let denoms = vec!["BTC", "mBTC", "uBTC", "nBTC", "pBTC", "bits", "sats", "msats"];
        for denom in denoms {
            for v in vec!["0", "-0", "000"] {
                let s = format!("{} {}", v, denom);
                match Amount::from_str(&s) {
                    Err(e) => panic!("Failed to crate amount from {}: {:?}", s, e),
                    Ok(amount) => assert_eq!(amount, Amount::from_sat(0)),
                }
            }
        }
    }
```

Reply:
You love vecs, don't you? ;)

Reply:
And Big-O n squared algorithms ;)